### PR TITLE
Made changes to Workflow Editor Product Tour based on latest design

### PR DESCRIPTION
**Pull request by cdhenley219-bit**
Changes are for story https://bitrise.atlassian.net/browse/PLG-318

### CODE REVIEW

File: source/javascripts/components/ProductTour/Highlighter.tsx
Comment (avivenzio-bitrise) [inline source/javascripts/components/ProductTour/Highlighter.tsx:56]: why do you need these offset values (`-10`) instead of passing in the desired value as it was in the previous implementation? Is it the addition of 10px border?

File: source/javascripts/components/ProductTour/Highlighter.tsx
Comment (avivenzio-bitrise) [inline source/javascripts/components/ProductTour/Highlighter.tsx:59]: same question as above

File: source/javascripts/components/ProductTour/ProductTooltip.tsx
Comment (avivenzio-bitrise) [inline source/javascripts/components/ProductTour/ProductTooltip.tsx:50]: kind of a small thing. wonder if its easier to read if the conditional is for the class name instead of changing the element like:
```tsx
<div className={`arrow-${tip.position === "right" ? "left" : "up"}`} />
```

File: source/javascripts/components/ProductTour/ProductTooltip.tsx
Comment (avivenzio-bitrise) [inline source/javascripts/components/ProductTour/ProductTooltip.tsx:44]: how does this behave on different screen sizes/zoom levels with these magic numbers?

Another thought is maybe pulling each out into a variable so when reading it might be easier to understand why the math is needed and what each number is based on.

**Review by avivenzio-bitrise — COMMENTED** [review]
The CI is failing because the e2e tests need the following update if you want to put iit in, not sure which pr will get merged first
https://github.com/bitrise-io/bitrise-workflow-editor/pull/712/files#diff-8038de0d16c18ffbd0581a0138debffc2a8ea7b2c3062c8fdc649e66a1ec738eR166

File: source/javascripts/components/ProductTour/Highlighter.tsx
Comment (cdhenley219-bit) [reply source/javascripts/components/ProductTour/Highlighter.tsx:56]: Yes, exactly. It helps to add the thick white border around each highlighted item.

File: source/javascripts/components/ProductTour/Highlighter.tsx
Comment (cdhenley219-bit) [reply source/javascripts/components/ProductTour/Highlighter.tsx:59]: Yes, this is all related to the same thing.

**Comment by cdhenley219-bit** [comment]
> The CI is failing because the e2e tests need the following update if you want to put iit in, not sure which pr will get merged first https://github.com/bitrise-io/bitrise-workflow-editor/pull/712/files#diff-8038de0d16c18ffbd0581a0138debffc2a8ea7b2c3062c8fdc649e66a1ec738eR166

Thanks for letting me know. If it's already implemented on your end, then I can wait until your PR is merged first. I'll put this one into Draft status until then.

File: source/javascripts/components/ProductTour/ProductTooltip.tsx
Comment (cdhenley219-bit) [reply source/javascripts/components/ProductTour/ProductTooltip.tsx]: Good catch! This has been updated.

File: source/javascripts/components/ProductTour/ProductTooltip.tsx
Comment (cdhenley219-bit) [reply source/javascripts/components/ProductTour/ProductTooltip.tsx:49]: Honestly, I hadn't thought much through it, as the page is already not super responsive. However, what I can do is have it default to the "bottom" position when in mobile view.

File: source/javascripts/components/ProductTour/ProductTooltip.tsx
Comment (roland-spekker-bitrise) [inline source/javascripts/components/ProductTour/ProductTooltip.tsx:54]: Now that we have a proper bitkit version, could we switch to the `Popover` component to fix a lot of manual stuff we do here? Render something for the `PopoverTrigger` and we could just delete most of the tooltip positioning related stuff. 

I would do it in a follow up PR if it's not trivial to do, to not interfere with this development.

File: source/javascripts/controllers/_MainController.js.erb
Comment (roland-spekker-bitrise) [inline source/javascripts/controllers/_MainController.js.erb:96]: I think the `viewModel.menuIds` were only used by the tour so we can delete them if different ids will be used

File: source/javascripts/components/ProductTour/ProductTooltip.tsx
Comment (cdhenley219-bit) [reply source/javascripts/components/ProductTour/ProductTooltip.tsx]: This has been taken out and replaced with the BitKit popover

File: source/javascripts/components/ProductTour/ProductTooltip.tsx
Comment (cdhenley219-bit) [reply source/javascripts/components/ProductTour/ProductTooltip.tsx]: This has been taken out and replaced with the BitKit popover

File: source/javascripts/components/ProductTour/ProductTooltip.tsx
Comment (cdhenley219-bit) [reply source/javascripts/components/ProductTour/ProductTooltip.tsx]: This has been updated. Please see my latest changes

File: source/javascripts/controllers/_MainController.js.erb
Comment (cdhenley219-bit) [reply source/javascripts/controllers/_MainController.js.erb:96]: These have been taken out. For some reason, when I tried doing it before, it resulted in unexpected behavior. However, it may have been due to other items during development, as it's now working perfectly without the old `menuId`s

**Review by avivenzio-bitrise — COMMENTED** [review]
looks good to me! just a few very small comments

**Review by roland-spekker-bitrise — CHANGES_REQUESTED** [review]

File: source/javascripts/components/ProductTour/HighlighterProductTooltip.tsx
Comment (avivenzio-bitrise) [reply source/javascripts/components/ProductTour/HighlighterProductTooltip.tsx:38]: good catch, we should add some linting rules around this.

Comment (cdhenley219-bit) [reply source/javascripts/components/ProductTour/HighlighterProductTooltip.tsx:38]: This was logic previously there that I reused, but I'll update it to follow this convention

Comment (roland-spekker-bitrise) [reply source/javascripts/components/ProductTour/HighlighterProductTooltip.tsx:38]: yep the logic was there but without a hook

File: source/javascripts/components/ProductTour/ProductTour.tsx
Comment (cdhenley219-bit) [reply source/javascripts/components/ProductTour/ProductTour.tsx]: These have been removed. It'd be nice if WFE had linting rules to check for things like this. Something to think about and maybe add in the near future.

File: source/javascripts/components/ProductTour/useHighlightedArea.ts
Comment (cdhenley219-bit) [reply source/javascripts/components/ProductTour/useHighlightedArea.ts]: Removed

File: source/javascripts/components/ProductTour/useHighlightedArea.ts
Comment (cdhenley219-bit) [reply source/javascripts/components/ProductTour/useHighlightedArea.ts]: I didn't receive any errors about it, likely because line 43 already has a check in place for the existence of the ref. Initially I made the update because I thought it was the root cause of the bug I was talking about in standup, but now that I know it isn't, I've reverted it back to the prior value.

File: source/javascripts/components/ProductTour/ProductTour.tsx
Comment (cdhenley219-bit) [reply source/javascripts/components/ProductTour/ProductTour.tsx]: This has been removed

File: source/javascripts/components/ProductTour/HighlighterProductTooltip.tsx
Comment (cdhenley219-bit) [reply source/javascripts/components/ProductTour/HighlighterProductTooltip.tsx]: Looking at it again, I'm not sure of what to change. `HighlighterProductTooltip` is a component, and I'm not making any calls to a hook within a conditional statement. I have moved the check for the null tip to where the other rendering logic is located to make it more readable, but if anything else is needed, please let me know.

**Review by avivenzio-bitrise — APPROVED** [review]

**Review by roland-spekker-bitrise — APPROVED** [review]

**Pull request merged**
